### DISCUSSION TITLE
json - as json vs as data?

### DISCUSSION THREAD

**Discussion by cfjedimaster** [humanwhocodes/humanfs#54] [General]
When I see the `.json` method mentions, the docs say "reads the given file and returns a JSON value.". 

I'm probably overthinking it, but when I see "JSON value", I see "a string that is valid json". Before I saw the doc I had assumed, "It's going to read the value, assume it's JSON, and then parse it, returning regular JavaScript data." 

So I guess my questions are - is it automatically parsing the JSON and if so, isn't it returning returning... I guess just 'regular' value (objects, arrays, whatever the JSON represents)?

**Comment by nzakas** [comment]
I'm using "JSON value" as noted in the [specification](https://www.json.org/json-en.html). So, it can be any value represented by JSON.

So yes, `json()` is automatically parsing JSON and returning it, just like `.json()` on the `Response` object (via `fetch()`).

**Reply by cfjedimaster** [reply]
Ok. I guess it just sounded weird to me. :) Like, part of me feels like if I tell you "i'm giving you JSON", it means I'm giving you a string, and that's how I read the doc. But it's parsed JSON, which is good, and maybe I'm only a weird thinker. ;)
